lib/u3d/downloader.rb

```diff
@@ -268,11 +268,23 @@ def download(version, cached_versions)
           file_name = UNITY_MODULE_FILE_REGEX.match(url)[1]
           file_path = File.expand_path(file_name, dir)
 
+          ini_file = INIparser.load_ini(version, [], os: :linux)
+          size = ini_file['size'] if ini_file
           # Check if file already exists
           # Note: without size or hash validation, the file is assumed to be correct
           if File.file?(file_path)
-            UI.important "File already downloaded at #{file_path}"
-            return file_path
+            if size
+              if File.size(file_path) != size
+                UI.important "File at #{file_path} is corrupted, deleting it"
```

Comment:
in the future we can try adding resume downloading...

See
```
curl -X HEAD -i http://beta.unity3d.com/download/b6e0e521da90/./unity-editor-installer-2017.2.0b2.sh
HTTP/1.1 200 OK
Server: Apache/2.4.10 (Debian)
Last-Modified: Fri, 07 Jul 2017 09:00:25 GMT
ETag: "aca042ad-553b676bffc40"
Accept-Ranges: bytes
Content-Length: 2896183981
Content-Type: text/x-sh
X-Backend: unityreleases
X-Varnish: 11153433
X-Varnish-Cache: MISS
Cache-Control: public, private, max-age=31536000
Date: Wed, 09 Aug 2017 17:14:09 GMT
Connection: keep-alive
Set-Cookie: SERVERID=reverse-proxy3; path=/
```

Let's keep it out for now